cmake/kokkos_tribits.cmake

```diff
@@ -307,8 +308,24 @@ FUNCTION(KOKKOS_INTERNAL_ADD_LIBRARY LIBRARY_NAME)
   )
 
   IF (KOKKOS_ENABLE_CUDA)
-    SET(LIB_cuda "-lcuda")
-    TARGET_LINK_LIBRARIES(${LIBRARY_NAME} PUBLIC cuda)
+    IF (KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+       SET(LIB_cuda "-lcuda -lcudart")
+       find_library( cuda_lib_ NAMES libcuda.so cuda HINTS ${KOKKOS_CUDA_DIR}/lib64 )
+       find_library( cudart_lib_ NAMES libcudart.so cudart HINTS ${KOKKOS_CUDA_DIR}/lib64 )
```

Comment:
Nice.  In my experience only the runtime library was missing on the link line.  Will need to check there is no duplication.

Reply:
Also a more general question: is it reasonable to rely on the [cache variable`KOKKOS_CUDA_DIR`](https://github.com/kokkos/kokkos/blob/4d7399b4c8accfd9f0eb796ed6647d75355f9971/cmake/kokkos_enable_options.cmake#L162) or should we attempt to locate the CUDA runtime library when it is not provided by the user.  I see the comment about defaulting to "where nvcc is installed" but I could not find the logic that would do that for Clang.

We could be look into doing something [along these lines](https://github.com/arborx/ArborX/blob/fae344746a3a9e3cd7be7e3366a7e64064abf6a5/cmake/FindKokkos.cmake#L90-L96) to help finding `cudart`.


(something like )